### ISSUE TITLE
Missing match arm false positive with conditionally enabled (#[cfg(feature = ...)]) enum variants

### ISSUE THREAD

**Issue by nickolay** [rust-analyzer/rust-analyzer#4279]
Using the current HEAD (2474f42ae95bffea7c0bc713f92322bfec4d59a7), I get a "missing match arm" diagnostic reported when matching on an `enum`, which declares one of its variants differently, depending on presence of a `cfg` feature. A minimal testcase looks like this:

```
    #[test]
    fn enum_feature() {
        let content = r#"
            enum Foo {
                #[cfg(not(feature = "myfeat"))]
                Bar,
                #[cfg(feature = "myfeat")]
                Bar,
                Baz
            }

            fn test_fn(f: Foo) {
                match f {
                    Foo::Bar => {},
                    Foo::Baz => {},
                }
            }
        "#;
        check_no_diagnostic(content);
    }
```

I guess this is related to PR #3706, so CCing @JoshMcguigan

**Comment by flodiebold** [comment]
I assume we just don't honor `#[cfg]` on enum variants yet, so it's not directly related to the diagnostic.

**Comment by bnjjj** [comment]
As I already implemented for struct field I will check to make the same for enum fields :)

**Comment by nickolay** [comment]
@bnjjj thanks for the speedy fix, works perfectly!
